.github/workflows/terraform-plan-apply.yml

```diff
@@ -302,6 +307,37 @@ jobs:
           az storage blob delete-batch --connection-string ${ui_storage_conn_string} --source "\$web" --pattern "*"
           az storage blob upload-batch --connection-string ${ui_storage_conn_string} --source mts-trial-ui --destination '$web'
 
+      # Download init service progress log file from azure file share and validate success
+      - name: Validate Init service progress
+        # if: github.ref == 'refs/heads/main'
+        run: |
+          #can't use errexit and pipefail flags here becasue tail will fail if the file doesn't exists
+          set -o nounset
+          #set -o xtrace
+
+          #exponential interval
+          for i in {1..10}; do
+            seconds=$(expr 30 \* $i)
+            echo "try #$i fetching init service log with a delay of $seconds seconds."
+            sleep $seconds
+            az storage file download --path log.txt --share-name ${init_storage_share_name} --connection-string ${init_storage_conn_string} &&             cat log.txt
+            result_failure=$(tail log.txt --lines 1 | grep "***FAILURE***" | wc -w)
+            result_success=$(tail log.txt --lines 1 | grep "***SUCCESS***" | wc -w)
+            echo result_failure=$result_failure
+            echo result_success=$result_success
+            if [ $result_failure -gt 0 ]; then
+                echo "Init service - Completed with failure"
+                exit 1
+            elif [ $result_success -gt 0 ]; then
+                echo "Init service - Completed Successfully"
+                exit 0
+            fi
+          done
+
+          # if we got here we need to fail with timeout
+          echo "quiting with timeout"
+          exit 1
```

Comment:
This step can be simpler - regular sleep (like 10 seconds) with a timeout on the step itself (not in bash)

Reply:
Correct, I've changed it to while-do loop based on github action timeout,
testing it right now